utils/asr_align_wav.sh

```diff
@@ -226,7 +231,13 @@ if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
     echo "$base $wav" > ${align_dir}/data/wav.scp
     echo "X $base" > ${align_dir}/data/spk2utt
     echo "$base X" > ${align_dir}/data/utt2spk
-    echo "$base $text" > ${align_dir}/data/text
+    if [ -f "$text" ]; then
+        cp "$text" "${align_dir}/data/text"
+        utt_text="$text"
+    else
+        echo "$base $text" > ${align_dir}/data/text
+        utt_text=${align_dir}/data/text
+    fi
```

Comment:
Changed in 761196bd7.

I remove ``-u`` from ``cp``.
This is because rerunning this script with a different old file will cause an error.
If using ``rsync`` instead of ``cp``, there is no problem.

I assing ``utt_text`` the original file, because the copied file will be truncated like this,
```console
$ cat ./egs/tedlium2/align1/data/demo/data/text
ctc_align_test THE SALE OF THE HOTELS
```

and the result file ``data/demo/aligned_segments`` only has one line.

Reply:
From your last commit:
```bash
utt_text="$text" # Use the original file, because copied file will be truncated
```
I'm not sure at which point the data is truncated. We don't want to discard any data, this would be a bug. How did this happen?

So, I'm not exactly sure what happened here, but with the "old" version, my `utt_text` was _not_ truncated and the `data.json` looks like this on my computer:
```json
/xxx/espnet_temp/egs/tedlium2/align1$ cat data/demo/dump/data.json 
{
    "utts": {
        "ctc_align_test": {
            "input": [
                {
                    "feat": "/xxx/espnet_temp/egs/tedlium2/align1/data/demo/dump/feats.1.ark:15",
                    "name": "input1",
                    "shape": [
                        652,
                        83
                    ]
                }
            ],
            "output": [
                {
                    "name": "target1",
                    "shape": [
                        18,
                        54
                    ],
                    "text": "data/demo/utt_text",
                    "token": "d a t a / d e m o / u t t _ t e x t",
                    "tokenid": "2 2 2 2 13 2 2 2 2 13 2 2 2 47 2 2 2 2"
                }
            ],
            "utt2spk": "X"
        }
    }
}
```

Reply:
>  I'm not sure at which point the data is truncated. We don't want to discard any data, this would be a bug. How did this happen?

The truncated file is ``egs/tedlium2/align1/data/demo/data/text``.
``egs/tedlium2/align1/data/demo/utt_text`` is not truncated.

```console
$  cat egs/tedlium2/align1/data/demo/data/text
ctc_align_test THE SALE OF THE HOTELS

$  cat egs/tedlium2/align1/data/demo/utt_text
ctc_align_test THE SALE OF THE HOTELS
ctc_align_test IS PART OF HOLIDAY'S STRATEGY
ctc_align_test TO SELL OFF ASSETS
ctc_align_test AND CONCENTRATE
ctc_align_test ON PROPERTY MANAGEMENT

```

# How I checked

```bash
cd egs/tedlium2/align1/
# data directory
align_dir=data/demo
rm -rf ${align_dir}
mkdir -p ${align_dir}
# wav file
base=ctc_align_test
wav=../../../test_utils/${base}.wav
# recipe files
echo "batchsize: 0" >${align_dir}/align.yaml

cat <<EOF >${align_dir}/utt_text
${base} THE SALE OF THE HOTELS
${base} IS PART OF HOLIDAY'S STRATEGY
${base} TO SELL OFF ASSETS
${base} AND CONCENTRATE
${base} ON PROPERTY MANAGEMENT
EOF

# pre-trained ASR model
model=wsj.transformer_small.v1
mkdir ./conf && cp ../../wsj/asr1/conf/no_preprocess.yaml ./conf

../../../utils/asr_align_wav.sh \
    --models ${model} \
    --align_dir ${align_dir} \
    --align_config ${align_dir}/align.yaml \
    ${wav} ${align_dir}/utt_text

cat data/demo/aligned_segments
```

``egs/tedlium2/align1/data/demo/aligned_segments`` should contain five lines.
By removing the following line in ``utils/asr_align_wav.sh`` (in the last commit 761196b)
```python
utt_text="$text" # Use the original file, because copied file will be truncated
```
``aligned_segments`` will contain only one line.

Reply:
Thanks for making that clear. The truncated file may be another issue with the script, as `espnet/bin/asr_align.py` only writes to a `segments` file...

I have one further small change request: Please put curly brackets around the shell variables, like so: `${utt_text}`. This is for consistency and good programming practice. Then we should be OK.

Reply:
Thank you.
There may be a problem with ``utils/fix_data_dir.sh``, but I don't know much about it.
I put curly brackets in 937b92341.

Reply:
Perfect!
Well, the script works, so we don't need to investigate further, right? :wink: 
Nevertheless, the pull request is ready for merge. Thank you for the pull request.